drupal-8/README.md

```diff
@@ -0,0 +1,155 @@
+  - Image field on Article content type (Structure > Content types > Article > Manage fields > Image)
+  - Image field for User profile picture (Configuration > People > Account settings > Manage fields > Picture)
+
+
+You are all set to use Drupal with Cloud Foundry\*. Congratulations!
+
+
+\*: see "Known Issues", below
+
+
+## Gotchas:
+
+1. `'v3-push' is not a registered command. See 'cf help'` : You'll need to update your CF CLI install.
+
+
+----
+# Building your own Drupal project
+
+This project demonstrates a Drupal project initiated with `composer`. Let's step through the process so you can repeat for your Drupal project. The steps are for MacOS. 
+
+Install composer:
+```
+brew install homebrew/php/composer
+```
+
+Create a fresh Drupal8 project named `d8ex`:
+```
+composer create-project drupal-composer/drupal-project:8.x-dev d8ex --stability dev --no-interaction
+cd d8ex
+composer require drupal/flysystem_s3
+```
+
+Check your work into Git:
+```
+git init
+git add .
+git commit -m "Initial commit from composer"
+```
+
+
+Customize for Cloud Foundry by copying the following to your project
+* Copy the `.bp-config/` directory to your project
+* `.cfignore`
+* `.gitignore` (we don't ignore the `settings.php` file)
+* `apt.yml`
+* `bootstrap.sh`
+* `manifest.yml`
+* `web/sites/default/settings.cf.php`
+
+
+E.g., if you've cloned this project to $HOME/Projects/18f/cf-ex-drupal/, then:
+
+```
+SOURCE_DRUPAL=$HOME/Projects/18f/cf-ex-drupal/drupal-8/
+cp -r $SOURCE_DRUPAL/.bp-config .
+git add .bp-config
+for f in .cfignore .gitignore apt.yml bootstrap.sh manifest.yml web/sites/default/settings.cf.php; do
+  cp $SOURCE_DRUPAL/$f $f
+  git add $f
+done
+```
+
+Add the service parsing to `settings.php` by pasting in the following:
+```
+if (file_exists($app_root . '/' . $site_path . '/settings.cf.php')) {
+  include $app_root . '/' . $site_path . '/settings.cf.php';
+}
+```
+
+Now you can push as you did above:
+
+
+# Debugging
+
+## SSH Setup
+
+If you `cf ssh` to the application, you'll need to set the following variables to work effectively with the DB or PHP:
+
+```
+export HOME=/home/vcap/app
+export DEPS_DIR=/home/vcap/deps
+export PYTHONPATH=/home/vcap/app/.bp/lib
+export TMPDIR=/home/vcap/tmp
+export LIBRARY_PATH=/home/vcap/deps/0/lib
+export PHPRC=/home/vcap/app/php/etc
+export PHP_INI_SCAN_DIR=/home/vcap/app/php/etc/php.ini.d
+export LD_LIBRARY_PATH=/home/vcap/deps/0/lib:/home/vcap/app/php/lib
+export PATH=/home/vcap/deps/0/bin:/usr/local/bin:/usr/bin:/bin:/home/vcap/app/php/bin:/home/vcap/app/php/sbin
+```
+
+## Development notes
+
+Use a dedicated mysql DB so it's easier to clean up without having to reprovision:
+
+
+# References
+
+These are complementary to what's described here for running on Heroku
+
+https://www.fomfus.com/articles/how-to-create-a-drupal-8-project-for-heroku-part-1
+https://www.fomfus.com/articles/how-to-deploy-a-drupal-8-project-to-heroku-part-2
+
+
+# Known issues
```

Comment:
I'd just remove things that are no longer unresolved... People don't need to see the interim steps if the end-result works. If you think the points you resolved are important, I'd hoist them out into a separate section that specifically talks about the points that were addressed and how/why. IOW, make this educational such that people can see how they might make similar decisions when approaching migrating their own app to CF.